sdk-extensions/autoconfigure/src/testFullConfig/java/io/opentelemetry/sdk/autoconfigure/FileConfigurationTest.java

```diff
@@ -183,4 +187,27 @@ void configFile_Error(@TempDir Path tempDir) throws IOException {
         .isInstanceOf(ConfigurationException.class)
         .hasMessage("Unrecognized span exporter(s): [foo]");
   }
+
+  @Test
+  void configFile_StructuredConfigProperties() {
+    ConfigProperties config =
+        DefaultConfigProperties.createFromMap(
+            Collections.singletonMap("otel.experimental.config.file", configFilePath.toString()));
+
+    AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk =
+        AutoConfiguredOpenTelemetrySdk.builder().setConfig(config).setResultAsGlobal().build();
+    OpenTelemetrySdk openTelemetrySdk = autoConfiguredOpenTelemetrySdk.getOpenTelemetrySdk();
+    cleanup.addCloseable(openTelemetrySdk);
+
+    // getConfig() should return ExtendedConfigProperties generic representation of the config file
+    StructuredConfigProperties structuredConfigProps =
+        autoConfiguredOpenTelemetrySdk.getStructuredConfig();
+    assertThat(structuredConfigProps).isNotNull();
+    StructuredConfigProperties otherProps = structuredConfigProps.getStructured("other");
+    assertThat(otherProps).isNotNull();
+    assertThat(otherProps.getString("str_key")).isEqualTo("str_value");
+    StructuredConfigProperties otherMapKeyProps = otherProps.getStructured("map_key");
+    assertThat(otherMapKeyProps).isNotNull();
+    assertThat(otherMapKeyProps.getString("str_key1")).isEqualTo("str_value1");
```

Comment:
This test demonstrates how a consumers like the otel java agent could access complex config data which is not part of the core file configuration schema. The `other` key and the structure it encloses is not part of the core schema yet its contents are accessible.

This will be important for configuring java agent specific options, like content for the JMX integration.

cc @trask, @laurit